src/field.rs

```diff
@@ -230,6 +230,15 @@ macro_rules! make_field {
         $elem:ident, $int:ident, $fp:ident, $encoding_size:literal, $encoding_order:expr,
     ) => {
         $(#[$meta])*
+        // This structure represents a field element in a prime order field. The concrete
+        // representation of the element is via the Montgomery domain. For an element n in GF(p),
+        // we store n * R^-1 mod p (where R is a given power of two). This representation enables
+        // using a more efficient multiplication algorithm, at the expense of having to convert
+        // elements between their Montgomery domain representation and natural representation.
+        // For calculations with many multiplications or exponentiations, this is worthwhile.
```

Comment:
Another reason for the choice of multiplication algorithm is to make it closer to constant time.